File: python/easy/two_sum_ii.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        ret = []
        if len(numbers) == 0:
            return ret
        start = 0
        end = len(numbers)-1
        while start < end:
            sum = numbers[start] + numbers[end]
            if sum == target:
                ret.append(start+1)
                ret.append(end+1)
                return ret
            elif sum < target:
                start += 1
            else:
                end -= 1

File: c++/easy/two_sum_ii.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ret;
        if(numbers.size() == 0)
            return ret;
        int start = 0;
        int end = numbers.size()-1;
        while(start < end)
        {
            int sum = numbers[start]+numbers[end];
            if(numbers[start]+numbers[end] == target)
            {
                ret.push_back(start+1);
                ret.push_back(end+1);
                return ret;
            }
            else if(sum > target)
                --end;
            else
                ++start;
        }
    }
};

Conclusion:
easy algorithm question, time complexity is O(n), space complexity is O(1)
